src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -281,11 +308,11 @@ private static bool IsValidIdentifier(string value)
         private JsonEncodedText FormatEnumValue(string value, JavaScriptEncoder? encoder)
         {
             Debug.Assert(_namingPolicy != null);
-            string formatted = FormatEnumValueToString(value, encoder);
+            string formatted = FormatEnumValueToString(value);
             return JsonEncodedText.Encode(formatted, encoder);
         }
 
-        private string FormatEnumValueToString(string value, JavaScriptEncoder? encoder)
+        private string FormatEnumValueToString(string value)
```

Comment:
Nit: consider renaming to something like `FormatUsingNamingPolicy`. This could potentially be a static method that accepts `NamingPolicy` as a parameter to avoid invalid invocations.